src/Bicep.Core/TypeSystem/VoidType.cs

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Bicep.Core.TypeSystem;
+
+public class VoidType : TypeSymbol
```

Comment:
How does `void` differ from `never`? Could you add a comment explaining the concept this represents?

Reply:
Looking at how we use `never` today, it's close enough to what I wanted to use `void` for that there's no reason we need both.